protomap/proto_test.go

```diff
@@ -73,20 +92,178 @@ func TestPathsFromProto(t *testing.T) {
+			mustPath("/list-name[single-key=k1]/another-field"):     "val-one",
+			mustPath("/list-name[single-key=k2]/single-key"):        "k2",
+			mustPath("/list-name[single-key=k2]/config/single-key"): "k2",
+			mustPath("/list-name[single-key=k2]/another-field"):     "val-two",
+		},
+	}, {
+		desc: "list with multiple keys",
+		inMsg: &epb.ExampleMessage{
+			Multi: []*epb.ExampleMessageMultiKey{{
+				Index: 0,
+				Name:  "zero",
+				Member: &epb.MultiKeyListMember{
+					Child: &wpb.StringValue{Value: "zero-child"},
+				},
+			}, {
+				Index: 1,
+				Name:  "one",
+				Member: &epb.MultiKeyListMember{
+					Child: &wpb.StringValue{Value: "one-child"},
+				},
+			}},
+		},
+		wantPaths: map[*gpb.Path]interface{}{
+			mustPath("/multi-list[index=0][name=zero]/index"):        uint32(0),
+			mustPath("/multi-list[index=0][name=zero]/config/index"): uint32(0),
+			mustPath("/multi-list[index=0][name=zero]/name"):         "zero",
+			mustPath("/multi-list[index=0][name=zero]/config/name"):  "zero",
+			mustPath("/multi-list[index=0][name=zero]/config/child"): "zero-child",
+			mustPath("/multi-list[index=1][name=one]/index"):         uint32(1),
+			mustPath("/multi-list[index=1][name=one]/config/index"):  uint32(1),
+			mustPath("/multi-list[index=1][name=one]/name"):          "one",
+			mustPath("/multi-list[index=1][name=one]/config/name"):   "one",
+			mustPath("/multi-list[index=1][name=one]/config/child"):  "one-child",
 		},
+	}, {
+		desc: "list with multiple paths",
+		inMsg: &epb.InvalidMessage{
+			Km: []*epb.ExampleMessageKey{{
+				SingleKey: "test",
+				Member: &epb.ExampleMessageListMember{
+					Str: &wpb.StringValue{Value: "failed"},
+				},
+			}},
+		},
+		wantErrSubstring: "invalid list, maps to >1 schema path",
+	}, {
+		desc: "repeated field that is not a list - unsupported",
+		inMsg: &epb.InvalidMessage{
+			Ke: []string{"one"},
+		},
+		wantErrSubstring: "invalid list, value is not a proto message",
+	}, {
+		desc: "list with bad key type",
+		inMsg: &epb.InvalidMessage{
+			Bk: []*epb.BadMessageKey{{
+				BadKeyType: 1.0,
+			}},
+		},
+		wantErrSubstring: "cannot map list key",
+	}, {
+		desc: "list with bad field type",
+		inMsg: &epb.InvalidMessage{
+			Bm: []*epb.BadMessageMember{{
+				Key:     "one",
+				BadType: []string{"one", "two"},
+			}},
+		},
+		wantErrSubstring: "invalid field in list",
+	}, {
+		desc: "invalid annotated path",
+		inMsg: &epb.InvalidMessage{
+			InvalidAnnotatedPath: &wpb.StringValue{Value: "one"},
+		},
+		wantErrSubstring: "received invalid annotated path",
+	}, {
+		desc: "invalid key names",
+		inMsg: &epb.InvalidMessage{
+			BkTwo: []*epb.BadMessageKeyTwo{{
+				Key: "one",
+			}},
+		},
+		wantErrSubstring: "received list key with leaf names that do not match",
 	}}
 
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {
-			got, err := pathsFromProto(tt.inMsg)
+			got := map[*gpb.Path]interface{}{}
+			err := pathsFromProtoInternal(tt.inMsg, got, tt.inBasePath)
```

Comment:
`inBasePath` doesn't seem to be tested by any test case. Is there a need to test it though? What about just testing `pathsFromProto()` directly instead of the internal function? That seems right to me.